internal/db/events.go

```diff
@@ -84,9 +84,8 @@ func Insert(ctx context.Context, dbc *sql.DB, key string, typ internal.EventType
 	}
 
 	notify, err := events.InsertWithMetadata(ctx, tx, key, typ, message)
-	if _, ok := MaybeWrapErrDuplicate(err, "by_type_key"); ok {
-		return err
-	} else if err != nil {
+	if err != nil {
+		err, _ = MaybeWrapErrDuplicate(err, "by_type_key")
```

Comment:
no need to check err!=nil first, if err == nil, ok will always be false
```
	if err, ok := MaybeWrapErrDuplicate(err, "by_type_key"); ok {
		return err
	}
```

Reply:
Updated. Please check this.